pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -676,6 +686,82 @@ func TestJavaInjection(t *testing.T) {
 				os.Remove(testfile)
 			},
 		},
+		{
+			// Test the java hotspot injection is working
+			name: "java_hotspot_injection_21_block_only",
+			context: testContext{
+				extras: make(map[string]interface{}),
+			},
+			preTracerSetup: func(t *testing.T, ctx testContext) {
+				ctx.extras["JavaAgentArgs"] = cfg.JavaAgentArgs
+
+				tfile, err := ioutil.TempFile(dir+"/../java/testdata/", "TestAgentLoaded.agentmain.*")
+				require.NoError(t, err)
+				tfile.Close()
+				os.Remove(tfile.Name())
+				ctx.extras["testfile"] = tfile.Name()
+				cfg.JavaAgentArgs += " testfile=/v/" + filepath.Base(tfile.Name())
+
+				// block the agent attachment
+				cfg.JavaAgentAllowRegex = ""
+				cfg.JavaAgentBlockRegex = ".*JustWait.*"
+			},
+			postTracerSetup: func(t *testing.T, ctx testContext) {
+				// if RunJavaVersion failing to start it's probably because the java process has not been injected
+				javatestutil.RunJavaVersion(t, "openjdk:21-oraclelinux8", "AnotherWait")
+				var fake testing.T
+				testSuccess := javatestutil.RunJavaVersion(&fake, "openjdk:21-oraclelinux8", "JustWait")
+				if testSuccess {
+					t.Fatalf("JustWait should not be attached")
+				}
+
+				cfg.JavaAgentArgs = ctx.extras["JavaAgentArgs"].(string)
+			},
+			validation: func(t *testing.T, ctx testContext, tr *Tracer) {
+				time.Sleep(time.Second)
+
+				testfile := ctx.extras["testfile"].(string)
+				_, err := os.Stat(testfile)
+				require.NoError(t, err)
+			},
+		},
+		{
```

Comment:
add a test to see that AllowRegex has higher priority by:
```go
cfg.JavaAgentAllowRegex = ".*JustWait.*"
cfg.JavaAgentBlockRegex = cfg.JavaAgentAllowRegex
```

Reply:
done